courses/serializers.py

```diff
@@ -195,6 +209,38 @@ def get_description(self, instance):
         """Description"""
         return instance.page.description if instance.page else None
 
+    def get_start_date(self, instance):
+        """Start date"""
+        return (
+            models.CourseRun.objects.filter(course__program=instance, live=True)
+            .order_by("start_date")
+            .values_list("start_date", flat=True)
+            .first()
+        )
+
+    def get_end_date(self, instance):
+        """End date"""
```

Comment:
nit: 
```
"""
end_date is the end date for the latest valid course run for all courses in a program.

:return end date.
"""
```

Reply:
I'll add more descriptive text to the docstring. However we are using sphinx-napoleon style docstrings: https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#docstrings

So that would be something like:

    Returns:
        datetime:
            The end date for the latest valid course run for all courses in a program